src/embedding.rs

```diff
@@ -369,13 +537,27 @@ where
             self.dimension, self.number_of_entities
         );
 
-        let result = M::init_with_hashes(
-            self.number_of_entities,
-            self.dimension,
-            self.fixed_random_value,
-            self.sparse_matrix_reader.clone(),
-        );
-
+        let result = match self.init_method {
+            InitMethod::Random => M::init_with_hashes(
+                self.number_of_entities,
+                self.dimension,
+                self.fixed_random_value,
+                self.sparse_matrix_reader.clone(),
+            ),
+            InitMethod::EvecSmallest => M::init_with_evec( // Is there a more compact solution?
```

Comment:
" // Is there a more compact solution?"

such a comment cannot be kept in release code. Please cleanup